index.md

```diff
@@ -1,217 +1,266 @@
+I am big fan of black box testing, especially when working with front-end. In the world of *javaScript frameworks changing so fast that before you download npm dependencies this dependency is already outdated* I cannot imagine writing tests that can be easily broken by the change in implementation that doesn't have any impact on the behavior of the application.  
 This kind of environment discourages developers from refactoring and writing tests because everything you touch has high chance of breaking something (be it tests or the actual code).
 
-We can avoid this problem by changing the way of thinking about our code and trying to test <i> behavior </i> and not the internal implementation (that can be easily changed). Doing so greatly reduces the need of changing tests when we want to change the way that our code works (e.g. refactoring code for optimization reasons).
+We can avoid this problem by changing the way of thinking about our code and trying to test *behavior* and not the internal implementation (that can be easily changed). Doing so greatly reduces the need of changing tests when we want to change the way that our code works (e.g. refactoring code for optimization reasons).
 
-//TODO: UMM PRETTY BAD EXAMPLE CONSIDERING THAT I AM WRITING ABOUT REUDX TESTING...
-For example, when testing let's say button counter component, I shouldn't care if the way of storing button counter is implemented within the component as internal state or Redux deals with it, this is an <i>implementation detail</i> that can change, but the <i> behavior </i> of the button will stay the same.
+`// TODO: UMM PRETTY BAD EXAMPLE CONSIDERING THAT I AM WRITING ABOUT REDUX TESTING...`  
+For example, when testing, let's say, button counter component, I shouldn't care if the way of storing button counter is implemented within the component as internal state or Redux deals with it, this is an *implementation detail* that can change, but the *behavior* of the button will stay the same.
 
 You might say, that this is a very basic example, real world is not that easy and components have some kind of dependencies (e.g. component is calling some endpoint to retrieve data), and how do I test that?
 
+`// FIXME: Please write what are yout trying to achieve. "black box testing" is too vague`  
+
 ## Test scenario
 
-Let's say that we want to test component that is responsible for managing movies, and for this basic example our component will have two methods: ``retrieveMovies()`` and ``clearMovies()``.
+*Note: All code samples can be found on my [github](https://github.com/Waksuu/react-exploratory/tree/after-redux-tests)*
+
+Let's say that we want to test component that is responsible for managing movies, and for this basic example our component will have two methods: `retrieveMovies()` and `clearMovies()`.
 
- `MoviePanel.component.tsx`
-```` typescript
+`MoviePanel.component.tsx`
+
+```typescript
 type Props = LinkStateProps & LinkDispatchProps;
 
+// FIXME: What is FC?
```

Comment:
It's a type provided by react and it represents a function, useful when you want to have types on your props